cai_causal_graph/time_series_causal_graph.py

```diff
@@ -111,6 +114,9 @@ def __init__(
             be created. If either or both are provided, but this is `False` (default), then the nodes will be added but
             not connected by edges.
         """
+        # Initialize caches for fast lookups
+        self.lag_to_nodes = defaultdict(list)
+        self.variable_name_to_nodes = defaultdict(list)
```

Comment:
These should both be protected attributes as the user should not be messing around with these themselves